### PR TITLE
add support for --timeout option

### PR DESCRIPTION
**Pull request by moshloop**
Harbor can take very long to lists tags on repos with 500+ tags

### CODE REVIEW

File: src/Harbor.Tagd/Args/CommonArgs.cs [hunk 1]
```diff
@@ -10,6 +10,9 @@ public class CommonArgs
 		[NamedArgument('v', "verbosity", Description = "How verbose should logging output be")]
 		public string Verbosity { get; set; } = "info";
 
+		[NamedArgument("timeout", Description = "Timeout in seconds for http requests, defaults to 600")]
+		public double Timeout { get; set; } = 600;
```
Comment (nlowe) [inline src/Harbor.Tagd/Args/CommonArgs.cs]: What's the reasoning behind making this a double instead of an int or a long?

Comment (moshloop) [reply src/Harbor.Tagd/Args/CommonArgs.cs]: TimeSpan.FromSeconds takes a double - I am not a C# developer so wasn't too sure on how type casting would work/apply.

Comment (nlowe) [reply src/Harbor.Tagd/Args/CommonArgs.cs]: No worries. Let's go ahead and make this an `int`. I don't think it makes sense to have a fractional-second timeout:

```suggestion
		public int Timeout { get; set; } = 600;
```

**Comment by nlowe** [comment]
@moshloop Thanks for the contribution! I'm looking into the build failures and hope to have a new version on Docker Hub later tonight.

**Comment by nlowe** [comment]
@moshloop This should be live in `hylandsoftware/tagd:1.7.2`

**Pull request merged**